### PR TITLE
feat: introduce interface name to NFMonitorMap

### LINKED ISSUE #201: Monitor map changes
Sample monitor map promQL format:

l3afd_NFMonitorMap{host="l3af-local-test",map_name="rl_drop_count_map_0_scalar",network_function="ratelimiting"} 1
l3afd_NFMonitorMap{host="l3af-local-test",map_name="rl_recv_count_map_0_max-rate",network_function="ratelimiting"} 21

1. Update monitor map with interface name tags.
2. Rename network_function to ebpf_program.

### PR DESCRIPTION
**Pull request by DecFox**
This diff introduces an `interface_name` field to `NFMonitorMap` and renames `network_function` to `ebpf_program`.
Closes #201

### CODE REVIEW

**Review by dthaler — DISMISSED** [review]

**Review by dthaler — DISMISSED** [review]

File: kf/bpf.go [hunk 1]
```diff
@@ -840,7 +840,7 @@ func (b *BPF) MonitorMaps(ifaceName string, intervals int) error {
 		}
 		bpfMap := b.MetricsBpfMaps[mapKey]
 		MetricName := element.Name + "_" + strconv.Itoa(element.Key) + "_" + element.Aggregator
-		stats.SetValue(bpfMap.GetValue(), stats.NFMointorMap, b.Program.Name, MetricName)
+		stats.SetValue(bpfMap.GetValue(), stats.NFMonitorMap, b.Program.Name, MetricName, "")
```
Comment (sanfern) [inline kf/bpf.go:843]: Can we pass interface name here, instead of empty string.

File: kf/processCheck.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func (c *pCheck) pMonitorWorker(bpfProgs map[string]*list.List, direction string
 						log.Error().Err(err).Msgf("pMonitor BPF Program start failed for program %s", bpf.Program.Name)
 					}
 				} else {
-					stats.Set(0.0, stats.NFRunning, bpf.Program.Name, direction)
+					stats.Set(0.0, stats.NFRunning, bpf.Program.Name, direction, "")
```
Comment (sanfern) [inline kf/processCheck.go:66]: Can we pass interface name here, instead of empty string.

File: kf/bpf.go [hunk 3]
```diff
@@ -237,10 +237,10 @@ func (b *BPF) Stop(ifaceName, direction string, chain bool) error {
 	// Reset ProgID
 	b.ProgID = 0
 
-	stats.Incr(stats.NFStopCount, b.Program.Name, direction)
+	stats.Incr(stats.NFStopCount, b.Program.Name, direction, "")
```
Comment (sanfern) [inline kf/bpf.go:240]: Can we pass interface name here, instead of empty string.

File: kf/bpf.go [hunk 4]
```diff
@@ -237,10 +237,10 @@ func (b *BPF) Stop(ifaceName, direction string, chain bool) error {
...
 
 	// Setting NFRunning to 0, indicates not running
-	stats.Set(0.0, stats.NFRunning, b.Program.Name, direction)
+	stats.Set(0.0, stats.NFRunning, b.Program.Name, direction, "")
```
Comment (sanfern) [inline kf/bpf.go:243]: Can we pass interface name here, instead of empty string.

File: kf/bpf.go [hunk 5]
```diff
@@ -434,8 +434,8 @@ func (b *BPF) Start(ifaceName, direction string, chain bool) error {
 	if err := b.SetPrLimits(); err != nil {
 		log.Warn().Err(err).Msg("failed to set resource limits")
 	}
-	stats.Incr(stats.NFStartCount, b.Program.Name, direction)
-	stats.Set(float64(time.Now().Unix()), stats.NFStartTime, b.Program.Name, direction)
+	stats.Incr(stats.NFStartCount, b.Program.Name, direction, "")
+	stats.Set(float64(time.Now().Unix()), stats.NFStartTime, b.Program.Name, direction, "")
```
Comment (sanfern) [inline kf/bpf.go:438]: Can we pass interface name here, instead of empty string.

File: kf/bpf.go [hunk 6]
```diff
@@ -462,7 +462,7 @@ func (b *BPF) Update(ifaceName, direction string) error {
 			bpfMap.Update(v)
 		}
 	}
-	stats.Incr(stats.NFUpdateCount, b.Program.Name, direction)
+	stats.Incr(stats.NFUpdateCount, b.Program.Name, direction, "")
```
Comment (sanfern) [inline kf/bpf.go:465]: Can we pass interface name here, instead of empty string.

**Review by jniesz — APPROVED** [review]
Thanks for contributing this and change LGTM. Can you please submit another PR for the dev environment to align to this PR and renaming of network_function --> ebpf_program?  

https://github.com/l3af-project/l3af-arch/search?q=network_function

**Comment by DecFox** [comment]
> Can you please submit another PR for the dev environment to align to this PR and renaming of network_function --> ebpf_program?

I have raised a PR here: https://github.com/l3af-project/l3af-arch/pull/61 which handles the required changes for the grafana dashboards.

**Pull request merged**